pennylane/ops/op_math/pow_class.py

```diff
@@ -0,0 +1,284 @@
+        copied_op = object.__new__(type(self))
+        # copied_op must maintain inheritance structure of self
+        # For example, it must keep AdjointOperation if self has it
+        # this way preserves inheritance structure
+
+        for attr, value in vars(self).items():
+            if attr != "_hyperparameters":
+                setattr(copied_op, attr, value)
+        copied_op._hyperparameters = copy(self._hyperparameters)
+        copied_op._hyperparameters["base"] = copy(self.base)
+
+        return copied_op
+
+    # pylint: disable=super-init-not-called
+    def __init__(self, base=None, z=0, do_queue=True, id=None):
+
+        # incorporate base inverse attribute into the exponent
+        if getattr(base, "inverse", False):
+            base.inverse = False
+            z *= -1
+
+        self.hyperparameters["base"] = base
+        self.hyperparameters["z"] = z
+        self._id = id
+        self.queue_idx = None
+
+        self._name = f"{self.base.name}**{z}"
+
+        if do_queue:
+            self.queue()
+
+    @property
+    def base(self):
+        """The operator that is raised to a power."""
+        return self.hyperparameters["base"]
+
+    @property
+    def z(self):
+        """The exponent."""
+        return self.hyperparameters["z"]
+
+    @property
+    def data(self):
+        """Trainable parameters that the operator depends on."""
+        return self.base.data
+
+    @data.setter
+    def data(self, new_data):
+        """Allows us to set base operation parameters."""
+        self.base.data = new_data
+
+    @property
+    def parameters(self):
+        return self.base.parameters
+
+    @property
+    def num_params(self):
+        return self.base.num_params
+
+    @property
+    def wires(self):
+        return self.base.wires
+
+    # pylint: disable=protected-access
+    @property
+    def _wires(self):
+        return self.base._wires
+
+    # pylint: disable=protected-access
+    @_wires.setter
+    def _wires(self, new_wires):
+        # used in a couple places that want to update the wires of an operator
+        # we should create a better way to set new wires in the future
+        self.base._wires = new_wires
+
+    @property
+    def num_wires(self):
+        return len(self.wires)
+
+    def queue(self, context=QueuingContext):
+        context.safe_update_info(self.base, owner=self)
+        context.append(self, owns=self.base)
+
+        return self
+
+    def label(self, decimals=None, base_label=None, cache=None):
+        z_string = format(self.z).translate(_superscript)
+        return self.base.label(decimals, base_label, cache=cache) + z_string
+
+    # pylint: disable=arguments-renamed, invalid-overridden-method
+    @property
+    def has_matrix(self):
+        return self.base.has_matrix
+
+    # pylint: disable=arguments-differ
+    @staticmethod
+    def compute_matrix(*params, base=None, z=0):
+        base_matrix = base.compute_matrix(*params, **base.hyperparameters)
```

Comment:
I would always use `base.matrix` from the outside. Or is there a special reason not to do that? 

In fact, in the prototype I overwrite `matrix` in the symbolic classes and found this quite nice because it takes care of params and hyperparams. Apologies if I overlooked this in the other PR.

Reply:
Basically, I didn't want to deal with `wire_order`.  I didn't want to copy and paste code over from the base class, because that makes the specifics of the implementation of a single decision exist in multiple places spread out through the code base.

But I'm switching to defining `matrix` instead.  I guess importing `expand_matrix` isn't that bad.